### ISSUE TITLE
yolov8-obb使用mosaic增强

### ISSUE THREAD

**Issue by zhibin-hong** [ultralytics/ultralytics#24616]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

训练参数： train_results = model.train(
        data=args.data,  # 数据集配置文件路径
        epochs=350,  # 训练周期数
        imgsz=640,  # 训练图像尺寸
        device=0,  # 运行设备（GPU 0）
        batch=32,  # 批次大小，可根据GPU内存调整
        patience=50,  # 早停机制
        save=True,  # 保存训练结果
        project="runs/train",  # 保存目录
        name=args.name,  # 名称
        agnostic_nms=False,
        multi_scale=True,
        mosaic=1,
        close_mosaic=45,
        #degrees=180.0,
        #hsv_h=0.015, hsv_s=0.4, hsv_v=0.2
    )
   我已知我的标签坐标是超出图片边缘，开启mosaic的时候，当旋转框跨越拼接边界时，它会被硬性截断。

### Environment

(ultralytics) hongzb@ps:~/code/PigConunter-Vision/yolo-pig-trainer$ /home/hongzb/miniconda3/envs/ultralytics/bin/python /home/hongzb/code/PigConunter-Vision/yolo-pig-trainer/evea/src/export.py
Ultralytics 8.3.40 🚀 Python-3.10.0 torch-2.7.1+cu126 CPU (Intel Core(TM) i9-14900K)
YOLO11s-obb summary (fused): 257 layers, 9,699,174 parameters, 0 gradients, 22.3 GFLOPs

PyTorch: starting from '/home/hongzb/code/PigConunter-Vision/yolo-pig-trainer/runs/train/allpig_yolov11-obb_20260526_argument_v52/weights/best.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 6, 8400) (19.3 MB)

### Minimal Reproducible Example

训练参数： train_results = model.train(
        data=args.data,  # 数据集配置文件路径
        epochs=350,  # 训练周期数
        imgsz=640,  # 训练图像尺寸
        device=0,  # 运行设备（GPU 0）
        batch=32,  # 批次大小，可根据GPU内存调整
        patience=50,  # 早停机制
        save=True,  # 保存训练结果
        project="runs/train",  # 保存目录
        name=args.name,  # 名称
        agnostic_nms=False,
        multi_scale=True,
        mosaic=1,
        close_mosaic=45,
        #degrees=180.0,
        #hsv_h=0.015, hsv_s=0.4, hsv_v=0.2
    )
   

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @zhibin-hong, thank you for reporting this and for sharing your training configuration for YOLO11-OBB 🚀 This is an automated response to help get the issue moving quickly, and an Ultralytics engineer will also assist soon.

We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many common questions may already be answered.

Since this appears to be a 🐛 bug report related to OBB + mosaic augmentation, please provide a complete [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if possible. In particular, it would help to include:
- A small sample image or two
- The corresponding OBB label files
- Your dataset YAML
- The exact training command or script used
- Any screenshots showing the truncation behavior when mosaic is enabled
- Confirmation whether this also occurs after upgrading to the latest package version

If this is a custom training ❓ question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify this is not already resolved in a newer release:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by zhibin-hong** [comment]
边缘部分

<img width="85" height="113" alt="Image" src="https://github.com/user-attachments/assets/6f0655ea-3200-4083-af35-635418d688d3" />

**Comment by glenn-jocher** [comment]
是的，这张图说明了当前实现里的边界裁剪：`Mosaic._update_labels()` 只做坐标平移，之后 `Mosaic._cat_labels()` 会对拼接后的 `instances` 调用 `clip()`，所以跨拼接边界的 OBB 会被直接截断，相关实现可参考 [Mosaic 文档](https://docs.ultralytics.com/reference/data/augment) 和 [`Instances.clip()`](https://docs.ultralytics.com/reference/utils/instance)；目前建议先关闭 `mosaic` 训练 OBB，或先把超出原图的标注裁到图内并删除 `labels` 下的 `*.cache` 后再重训。

**Comment by zhibin-hong** [comment]
感谢您的回答，建议直接裁切到图像内吗？但是这么做的话标注是不完整的目标，这样会对训练有影响吗

**Comment by zhibin-hong** [comment]
<img width="207" height="160" alt="Image" src="https://github.com/user-attachments/assets/868bae2f-36b8-4487-bdb3-8ab2b7d58934" />是才切成绿色的四边形（直角梯形），还是红色的旋转矩形

**Comment by glenn-jocher** [comment]
如果是原图就被边界截断的目标，建议标成红色的旋转矩形，只保留图内可见部分；不要标绿色梯形，因为 Ultralytics YOLO 的 OBB 是 4 个角点定义的旋转框，内部按 `xywhr` 处理，不是任意四边形，参考 [OBB 数据集格式](https://docs.ultralytics.com/zh/datasets/obb) 和 [OBB 任务说明](https://docs.ultralytics.com/tasks/obb)。 ([docs.ultralytics.com](https://docs.ultralytics.com/zh/datasets/obb)) 少量这类截断样本一般可以接受，关键是全数据集标注规则保持一致；但如果主要是被 `mosaic` 拼接边界截出来的“假截断”，还是更建议直接关闭 `mosaic`。
